core/opendaq/reader/src/block_reader_impl.cpp

```diff
@@ -129,32 +159,35 @@ ErrCode BlockReaderImpl::packetReceived(IInputPort* inputPort)
 
 ErrCode BlockReaderImpl::readPacketData()
 {
-    auto remainingSampleCount = info.dataPacket.getSampleCount() - info.prevSampleIndex;
-    SizeT toRead = std::min(remainingSampleCount, info.remainingSamplesToRead);
+    auto packetSampleCount = info.currentDataPacketIter->getSampleCount();
+    auto packetRemainingSampleCount = packetSampleCount - info.prevSampleIndex;
+    auto blockRemainingSampleCount = blockSize - info.writtenSampleCount % blockSize;
+    SizeT sampleCountToRead = std::min(blockRemainingSampleCount, packetRemainingSampleCount);
 
-    ErrCode errCode = valueReader->readData(getValuePacketData(info.dataPacket), info.prevSampleIndex, &info.values, toRead);
+    auto* packetData = getValuePacketData(*info.currentDataPacketIter);
+    ErrCode errCode = valueReader->readData(packetData, info.prevSampleIndex, &info.values, sampleCountToRead);
     if (OPENDAQ_FAILED(errCode))
     {
         return errCode;
     }
-
     if (info.domainValues != nullptr)
     {
-        auto dataPacket = info.dataPacket;
+        auto dataPacket = *info.currentDataPacketIter;
         if (!dataPacket.getDomainPacket().assigned())
         {
             return makeErrorInfo(OPENDAQ_ERR_INVALIDSTATE, "Packets must have an associated domain packets to read domain data.");
         }
 
         auto domainPacket = dataPacket.getDomainPacket();
-        errCode = domainReader->readData(domainPacket.getData(), info.prevSampleIndex, &info.domainValues, toRead);
+        auto* domainData = domainPacket.getData();
+        errCode = domainReader->readData(domainData, info.prevSampleIndex, &info.domainValues, sampleCountToRead);
         if (errCode == OPENDAQ_ERR_INVALIDSTATE)
         {
             if (!trySetDomainSampleType(domainPacket))
             {
                 return errCode;
             }
-            errCode = domainReader->readData(domainPacket.getData(), info.prevSampleIndex, &info.domainValues, toRead);
+            errCode = domainReader->readData(domainPacket.getData(), info.prevSampleIndex, &info.domainValues, sampleCountToRead);
```

Comment:
Could use the `domainData` variable instead of calling `getData()`

Reply:
Done